pkg/api/message/v1/service.go

```diff
@@ -167,6 +168,58 @@ func (s *Service) Subscribe(req *proto.SubscribeRequest, stream proto.MessageApi
 	}
 }
 
+func (s *Service) Subscribe2(stream proto.MessageApi_Subscribe2Server) error {
+	log := s.log.Named("subscribe2")
+	log.Debug("started")
+	defer log.Debug("stopped")
+	// Send a header (any header) to fix an issue with Tonic based GRPC clients.
+	// See: https://github.com/xmtp/libxmtp/pull/58
+	_ = stream.SendHeader(metadata.Pairs("subscribed", "true"))
+
+	ch := make(chan interface{})
+	defer s.dispatcher.Unregister(ch)
+
+	requestChannel := make(chan *proto.SubscribeRequest)
+	go func() {
+		for {
+			req, err := stream.Recv()
+			if err != nil {
+				if err != io.EOF && err != context.Canceled {
+					log.Error("reading subscription", zap.Error(err))
+				}
+				close(requestChannel)
+				return
+			}
+			requestChannel <- req
+		}
+	}()
+
+	for {
+		select {
+		case <-stream.Context().Done():
+			log.Debug("stream closed")
+			return nil
+		case <-s.ctx.Done():
+			log.Info("service closed")
+			return nil
+		case req := <-requestChannel:
+			log.Info("updating subscription", zap.Int("num_content_topics", len(req.ContentTopics)))
```

Comment:
I'm not logging the full list of topics in this new method. We should think about removing that in the main subscription method too. The subscription list is sometimes thousands of items long, and has little value in the logs.

Reply:
Would it make sense to log the topics added/removed instead?

Reply:
Still would be a very long list for the initial call of a popular user/bot